core/src/main/java/io/questdb/griffin/engine/groupby/AbstractNoRecordSampleByCursor.java

```diff
@@ -197,12 +208,25 @@ protected void initTimestamps() {
             // this is the default path, we align time intervals to the first observation
             timestampSampler.setStart(timestamp);
         } else {
-            timestampSampler.setStart(fixedOffset != Long.MIN_VALUE ? fixedOffset : 0L);
+            // FROM-TO may apply to align to calendar queries, fixing the lower bound.
+            if (sampleFromFunc != TimestampConstant.NULL) {
+                timestampSampler.setStart(fixedOffset != Long.MIN_VALUE ? sampleFromFunc.getTimestamp(null) : 0L);
```

Comment:
Shouldn't we add the offset to the FROM value? I'd expect it to apply to FROM/TO.

Reply:
I've suggested doing this since we use `fixedOffset` as the start timestamp when `sampleFromFunc != TimestampConstant.NULL`, so it would be consistent to use it when the FROM value is specified. On the other hand, I agree that in both cases the offset won't apply to the WHERE clause so the overall behavior may be unexpected for the user. In any case, we should document how things are working, so it doesn't come as a surprise for users. WDYT?

Reply:
Sounds good!